// BatchingRPCServer
// Copyright (c) 2017 Aman LaChapelle
// Full license at BatchingRPCServer/LICENSE.txt
//

/*
    Licensed under the Apache License, Version 2.0 (the "License");
    you may not use this file except in compliance with the License.
    You may obtain a copy of the License at

        http://www.apache.org/licenses/LICENSE-2.0

    Unless required by applicable law or agreed to in writing, software
    distributed under the License is distributed on an "AS IS" BASIS,
    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
    See the License for the specific language governing permissions and
    limitations under the License.
 */

syntax = "proto3";

package Serving;

message TensorMessage {
    repeated float buffer = 1[packed=true];
    int32 n = 2;
    int32 k = 3;
    int32 nr = 4;
    int32 nc = 5;
    string name = 6;
}

message ConnectionRequest {
//    bool first_request = 1;
}

message ConnectionReply {
    string client_id = 1;
}

/*
    The protocol is:
     - Send Connect call
     - Receive your uuid back. Add this to the metadata of every message you send thereafter.
*/
service BatchingServable {
    rpc Connect(ConnectionRequest) returns (ConnectionReply) {}
    rpc Process (TensorMessage) returns (TensorMessage) {} // async call
}